### PR TITLE
Smart Hub 2 Support

### PR DESCRIPTION
**Pull request by goul**
Added smarthub 2 support.

Simple attempt to detect a smarthub 2 vs a smarthub 1. 

Changed signature to :

`def get_devicelist(router_ip='192.168.1.254', only_active_devices=False, is_smarthub2=None):
`
Leaving is_smarthub2 as None means code will attempt to hit what I think is a v2 only URI to detect. If caller knows its a hub 2 (or 1) then we avoid doing the poll each time (I'm not sure how state should be stored after first lookup - in my world that would be a static member var).

Very happy to learn where I'm going wrong - very much a Java dev by trade, so I may be doing terrible things in Python.

### CODE REVIEW

File: bt_smarthub_devicelist/btsmarthub_devicelist.py [hunk 1]
```diff
@@ -5,16 +5,33 @@
 import random
 import hashlib
 import logging
+from bt_smarthub_devicelist import btsmarthub2_devicelist
```
Comment (MikhailMS) [inline bt_smarthub_devicelist/btsmarthub_devicelist.py:8]: I believe you do have a typo here
it should be 
```from btsmarthub_devicelist import btsmarthub2_devicelist``` instead :)

File: bt_smarthub_devicelist/btsmarthub2_devicelist.py [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+    request_url = 'http://' + router_ip + '/cgi/cgi_basicMyDevice.js'
+    _LOGGER.error("Scanning for home hub2 at %s", router_ip)
+    try:
+        response = requests.get(request_url, timeout=wait_time)
+    except requests.exceptions.Timeout:
+        _LOGGER.error("Connection to the router (%s) times out after %fs ", router_ip, wait_time)
+        return False
+
+    if response.status_code == 200:
+        _LOGGER.error("Router (%s) appears to be a Smart Hub 2 ", router_ip)
```
Comment (MikhailMS) [inline bt_smarthub_devicelist/btsmarthub2_devicelist.py:19]: Seems to me that probably you wanted this one to be 
```_LOGGER.info("Router (%s) appears to be a Smart Hub 2 ", router_ip)```

**Comment by MikhailMS** [comment]
Hope this one could be merged soon, cause HA fails for me without this release in place

I've added this patch manually and so far it works nicely :) Great job @paulgoulbourn @goul

File: bt_smarthub_devicelist/btsmarthub_devicelist.py (hunk 1 above)
Comment (d0ugal) [reply bt_smarthub_devicelist/btsmarthub_devicelist.py:8]: I fixed this in #8. The package name was refactored after this pull request was opened.

Comment (MikhailMS) [reply bt_smarthub_devicelist/btsmarthub_devicelist.py:8]: Oh, haven't seen that. Sorry :) looks like support for Smart Hub is happening soon

**Pull request closed without merging**